lizard_languages/clike.py

```diff
@@ -135,7 +135,7 @@ def _state_global(self, token):
         if token == "template":
             self._state = self._template_declaration
 
-        elif token in ("struct", "class", "namespace"):
+        elif self.last_token != "." and token in ("struct", "class", "namespace"):
             self._state = self._read_namespace
```

Comment:
This seems to be a separate issue from the fixing of division by zero.
I would suggest to put it into another PR.